internal/testing/pruning_test.go

```diff
@@ -35,41 +35,50 @@ func TestHostPruning(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// remove the first host manually and close it
-	cluster.hosts = cluster.hosts[1:]
-	err = hosts[0].Close()
-	if err != nil {
-		t.Fatal(err)
-	}
-
 	// add h1 to the blocklist
 	hk1 := hosts[0].PublicKey()
 	err = b.UpdateHostBlocklist(ctx, []string{hk1.String()}, nil, false)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	// record a couple of failed interactions for h1 that push his downtime well
-	// over the default 'MaxDowntimeHours' setting of 10 hours, and his recent
-	// scan failures over the default 'MinRecentScanFailures'
-	//
-	// NOTE: we do this manually to avoid bypassing the condition by adding
-	// flags or extending the config with properties only used in testing
-	his := make([]hostdb.Interaction, 10)
-	for i := 0; i < 10; i++ {
-		his[i] = hostdb.Interaction{
-			Host:      hk1,
-			Timestamp: time.Now().Add(time.Hour * 20),
-			Type:      hostdb.InteractionTypeScan,
-		}
+	// assert the host was scanned
+	h, err := b.Host(context.Background(), hk1)
+	if err != nil {
+		t.Fatal(err)
 	}
-	if err = b.RecordInteractions(context.Background(), his); err != nil {
+	if h.Interactions.LastScan.IsZero() {
+		t.Fatal("expected last scan to be set")
+	}
+
+	// remove the first host manually and close it
+	cluster.hosts = cluster.hosts[1:]
+	err = hosts[0].Close()
+	if err != nil {
 		t.Fatal(err)
 	}
 
 	// wait until we have 2 hosts in the set - we expect one host to be removed
 	// by the host pruning - this takes ~10 failed scans so we retry for 30s
-	if err := build.Retry(30, time.Second, func() error {
+	now := time.Now()
+	if err := Retry(30, time.Second, func() error {
```

Comment:
Is there a reason this needs to take 30s? If you know the exact number of interactions required why not rapidly record them in a loop and then wait for a few seconds?

Reply:
There's no reason for this to be 30s, but there's no reason for it to be more strict either? If the autopilot is involved, meaning you don't know when you might expect something to happen, I usually prefer to check in a `Retry` over sleeping for x seconds because the x would be the worst case so your test always takes worst-case-seconds.